src/probnum/filtsmooth/gaussfiltsmooth/kalmanposterior.py

```diff
@@ -132,68 +199,85 @@ def interpolate(self, t):
+        # the shape of base_measure_realization is not (len(locations), shape(RV))
+        # t_shape = self.locations.shape if t is None else (len(t) + 1,)
+        size_zero_shape = () + t.shape + self.states[0].shape
+        if base_measure_realizations.shape != size_zero_shape:
             return np.array(
-                self.transition.jointly_sample_list_backward(
-                    locations=locations, rv_list=random_vars
-                )
+                [
+                    self.transform_base_measure_realizations(
+                        base_measure_realizations=base_real,
+                        t=t,
+                    )
+                    for base_real in base_measure_realizations
+                ]
             )
 
+        # The final location is contained in  't' if this function is called from sample().
+        # If `transform_base_measure_realizations()` is called directly from the outside,
+        # you better know what you're doing ;)
+        rv_list = self.filtering_posterior(t)
         return np.array(
-            [self.sample(locations=locations, size=size[1:]) for _ in range(size[0])]
+            self.transition.jointly_transform_base_measure_realization_list_backward(
+                base_measure_realizations=base_measure_realizations,
+                t=t,
+                rv_list=rv_list,
+            )
         )
 
 
 class FilteringPosterior(KalmanPosterior):
-    """Filtering posterior.
+    """Filtering posterior."""
 
-    Parameters
-    ----------
-    locations : `array_like`
-        Locations / Times of the discrete-time estimates.
-    state_rvs : :obj:`list` of :obj:`RandomVariable`
-        Estimated states (in the state-space model view) of the discrete-time estimates.
-    transition : :obj:`Transition`
-        Dynamics model used as a prior for the filter.
-    """
+    def interpolate(self, t: DenseOutputLocationArgType) -> DenseOutputValueType:
+        """Predict to the present point.
 
-    def interpolate(self, t):
-        """Predict to the present point."""
+        Parameters
+        ----------
+        t :
+            Location to evaluate at.
+
+        Returns
+        -------
+        random_variables.RandomVariable or _randomvariablelist._RandomVariableList
+            Dense evaluation.
+        """
         previous_idx = self._find_previous_index(t)
         previous_t = self.locations[previous_idx]
-        previous_rv = self.state_rvs[previous_idx]
+        previous_rv = self.states[previous_idx]
 
         rv, _ = self.transition.forward_rv(previous_rv, t=previous_t, dt=t - previous_t)
         return rv
 
-    def sample(self, locations=None, size=()):
+    def sample(
+        self,
+        t: Optional[DenseOutputLocationArgType] = None,
+        size: Optional[ShapeArgType] = (),
+        random_state: Optional[RandomStateArgType] = None,
+    ) -> np.ndarray:
         raise NotImplementedError
```

Comment:
I dont think this is the full point. The superclass implements `sample()` by delegationg to `transform_base_measure_...` and when trying to sample from the fitlering posterior, this is where the error would be thrown. I thought it would be a bit more human-readable if the error was thrown right at sample(), which is why it is overwritten here.
However, now that we're talking about this, it should have an error message. 

What are your thoughts?

Reply:
Ah I see, makes sense. Yes I agree, an error message would be useful

Reply:
Done